### PR TITLE
Added changelog for provider version V1.4.1

### PR DESCRIPTION
**Pull request by a-atri**
## Jira

* AV-93122

## Description

Added Changelog for provider V1.4.1

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [ ] Manually tested
- [ ] Unit tested
- [ ] Acceptance tested
- [x] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>
  
</details>

## Required Checklist:

- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if required)
- [x] I have run make fmt and formatted my code
- [x] I have made sure that no schema field is marked with both requiresReplace and computed
## Further comments

### CODE REVIEW

**Review by simon-dew — DISMISSED** [review]
Minor comment

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
 # Changelog
 
+## [v1.4.1](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/v1.4.1) (2024-12-20)
+
+[Full Changelog](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/compare/v1.4.0...v1.4.1)
+
+**Implemented enhancements:**
+- Allows setting the provider host and authorization_token via CAPELLA prefixed environment variables [\#239](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/239) ([cdsre](https://github.com/cdsre))
```
Comment (Rebecca-Martinez007) [inline CHANGELOG.md:8]:
```suggestion
- Allows the provider `host` and and `authentication_token` to be set through environment variables prefixed with `CAPELLA_HOST` and `CAPELLA_AUTH_TOKEN`. [\#239](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/239) ([cdsre](https://github.com/cdsre))
```

Comment (Rebecca-Martinez007) [reply CHANGELOG.md:8]: Similar to what @simon-dew suggested and after going through the PR linked here, I set up the variables as written in the PR.

- `authorization_token` is actually `authentication_token` in the PR.
- The CAPELLA environment variables in the PR are written as `CAPELLA_HOST` and `CAPELLA_AUTH_TOKEN`. 

I just want to confirm that these variable are still written as this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+
+**Fixed bugs:**
+- \[AV-92771\] Fix GSI documentation link  [\#248](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/248) ([a-atri](https://github.com/a-atri))
+
+**Documentation Enhancement:**
+- \[AV-92775\] Update getting started GSI example to create non deferred secondary index [\#247](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/247) ([l0n3star](https://github.com/l0n3star))
```
Comment (Rebecca-Martinez007) [inline CHANGELOG.md:14]:
```suggestion
- \[AV-92775\] Update the Getting Started GSI example to create a non-deferred secondary index [\#247](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/247) ([l0n3star](https://github.com/l0n3star))
```

**Review by Rebecca-Martinez007 — CHANGES_REQUESTED** [review]
Just a few comments :) Let me know if there are any issues!

**Review by SaicharanCB — DISMISSED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+- \[AV-92771\] Fix GSI documentation link  [\#248](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/248) ([a-atri](https://github.com/a-atri))
+
+**Documentation Enhancement:**
+- \[AV-92775\] Update the Getting Started GSI example to create a non-deferred secondary index [\#247](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/247) ([l0n3star](https://github.com/l0n3star))
+
+**Merged pull requests:**
+- Bump github.com/stretchr/testify from 1.9.0 to 1.10.0 [#251](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/251) ([dependabot[bot]](https://github.com/apps/dependabot))
+- Bump golang.org/x/crypto from 0.30.0 to 0.31.0 [#249](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/249) ([dependabot[bot]](https://github.com/apps/dependabot))
+- Bump github.com/couchbase/tools-common/functional from 1.2.0 to 1.3.1 [#246](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/246) ([dependabot[bot]](https://github.com/apps/dependabot))
+
```
Comment (PaulomeeCb) [inline CHANGELOG.md:20]: @a-atri Even this dependabot PR was merged - https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/237

Comment (a-atri) [reply CHANGELOG.md:21]: @PaulomeeCb I have addressed your comment

**Review by SaicharanCB — APPROVED** [review]

**Review by PaulomeeCb — APPROVED** [review]
LGTM

**Pull request merged**